rest/config_flags.go

```diff
@@ -0,0 +1,198 @@
+package rest
+
+import (
+	"flag"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/hashicorp/go-multierror"
+)
+
+// configFlag stores the config value, and the corresponding flag value
+type configFlag struct {
+	config    interface{}
+	flagValue interface{}
+}
+
+// registerConfigFlags holds a map of the flag values with the configFlag it goes with
+// (which stores the corresponding config field pointer and flag value).
+func registerConfigFlags(config *StartupConfig, fs *flag.FlagSet) map[string]configFlag {
+	return map[string]configFlag{
```

Comment:
I'm wondering how we can catch developer-time errors like adding a new config option but forgetting to update this map, or vice versa.

Is it possible for a test to count all of the fields (including children) in StartupConfig and ensure it matches the number of elements returned by this function? That at least gives us an early-warning if we forget to update flags.